src/main/java/com/samsolutions/employeesdep/model/services/impl/EmployeeServiceImpl.java

```diff
@@ -0,0 +1,129 @@
+    @Transactional
+    public List<EmployeeDTO> getEmployeesToDepartment(String departName, int page) {
+        return getEmployeesToDepartmentCommon(departName, page, employeesPerPage);
+    }
+
+    @Override
+    @Transactional
+    public List<EmployeeDTO> getAllEmployees() {
+        Pageable paging = PageRequest.of(0, employeesPerPage);
+        EmployeeEntityToDTOConverter employeeEntityToDTOConverter = new EmployeeEntityToDTOConverter();
+        return empRepository.findAll(paging).stream() //stream from list
+                .map(employeeEntityToDTOConverter::convert) //convert every element
+                .collect(Collectors.toList());
+    }
+
+    @Override
+    @Transactional
+    public List<EmployeeDTO> getAllEmployees(int page) {
+        Pageable paging = PageRequest.of(page, employeesPerPage);
+        EmployeeEntityToDTOConverter employeeEntityToDTOConverter = new EmployeeEntityToDTOConverter();
+        return empRepository.findAll(paging).stream() //stream from list
+                .map(employeeEntityToDTOConverter::convert) //convert every element
+                .collect(Collectors.toList());
+    }
+
+    @Override
+    @Transactional
+    public EmployeeDTO createEmployee(EmployeeDTO employeeToSaveDTO) {
+        Employee employeeToSave = new EmployeeDTOToEntityConverter().convert(employeeToSaveDTO);
+        if (employeeToSave == null)
+            return null;
+
+        // saving department if new
+        Department departToSave;
+        if (employeeToSaveDTO.getDepartment().getId() == null) {
+            departToSave = new Department(employeeToSaveDTO.getDepartment().getName());
+            departRepository.save(departToSave);
+        } else {
+            departToSave = departRepository.getReferenceById(employeeToSaveDTO.getDepartment().getId());
+        }
+        employeeToSave.setDepartment(departToSave);
+
+        // saving roles if new
+        RoleDTOToEntityConverter roleDTOToEntityConverter = new RoleDTOToEntityConverter();
+        Set<Role> rolesToSave = new HashSet<>();
+        for (RoleDTO roleDTO : employeeToSaveDTO.getEmployeeRoles()) {
+            Role roleToSave = roleDTOToEntityConverter.convert(roleDTO);
+            if (roleToSave != null) {
+                if (roleToSave.getId() == null || roleToSave.getId() == 0) {
+                    roleRepository.save(roleToSave);
+                }
+                rolesToSave.add(roleToSave);
+            }
+        }
+        employeeToSave.setEmployeeRoles(rolesToSave);
+
+        // saving employee
+        empRepository.save(employeeToSave);
+        // convert employee entity back to DTO to have all IDs
+        return new EmployeeEntityToDTOConverter().convert(employeeToSave);
+    }
+
+    @Override
+    @Transactional
+    public void deleteEmployeeById(Long employeeId) {
+        if (empRepository.existsById(employeeId))
+            empRepository.deleteById(employeeId); // how to know whether delete was successful???
```

Comment:
how to know whether delete was successful??? - you may search for it after removal and check it is removed